web/modules/custom/solution/solution.module

```diff
@@ -15,3 +17,38 @@ function solution_entity_bundle_field_info_alter(&$fields, EntityTypeInterface $
     $fields['label']->addConstraint('UniqueSolutionTitle', []);
   }
 }
+
+/**
+ * Implements hook_entity_insert().
+ *
+ * The solution is connected to the collection with the affiliation field in the
+ * collection entity. The process of linking a solution to a collection is:
+ *  - Request an affiliation. This request is registered by adding the
+ *    collection to the affiliation_request field of the solution.
```

Comment:
This documentation is not intended for the end user. The end user should not have immediate interaction with these two fields described here.
The purpose of this documentation, is for the developer to know that due to the ADMS schema, the request is saved in the solution but the affiliation itself is saved in the collection.

I also think that this should be done merely by action buttons through the UI. The end user would never manage to actually go through that process manually, nor he would have permissions to do it (edit a solution to request an affiliation by filling his own collections name there)